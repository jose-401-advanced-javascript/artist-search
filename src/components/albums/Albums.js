import React from 'react';
import PropTypes from 'prop-types';
import AlbumCard from './AlbumCard';

const Albums = ({ albums, incrementPage, decrementPage, name }) => {

  const albumElements = albums.map(album => {
    return (
      <li key={album.id}>
        <AlbumCard {...album} name={name} />
      </li>
    );
  });

  return (
    <>
      <button type="button" onClick={decrementPage}>Back</button>
      <button type="button" onClick={incrementPage}>Next</button>
      <ul>
        {albumElements}
      </ul>
    </>
  );
};

Albums.propTypes = {
  albums: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    image: PropTypes.string,
  })),
  name: PropTypes.string,
  incrementPage: PropTypes.func,
  decrementPage: PropTypes.func
};

export default Albums;
