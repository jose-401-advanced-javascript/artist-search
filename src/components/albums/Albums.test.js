import React from 'react';
import { shallow } from 'enzyme';
import Albums from './Albums';

describe('Albums component', () => {
  it('renders Albums', () => {
    const wrapper = shallow(<Albums albums={[
      {
        'title': 'Liar',
        'quality': 'normal',
        'release-events': [
          {
            'area': {
              'name': 'United States',
              'sort-name': 'United States',
              'disambiguation': '',
              'iso-3166-1-codes': [
                'US'
              ],
              'id': '489ce91b-6658-3307-9877-795b68554c98'
            },
            'date': '1974-02-14'
          }
        ],
        'status-id': '4e304316-386d-3409-af2e-78857eec5cfe',
        'packaging': 'Cardboard/Paper Sleeve',
        'date': '1974-02-14',
        'disambiguation': '',
        'cover-art-archive': {
          'front': false,
          'darkened': false,
          'count': 2,
          'artwork': true,
          'back': false
        },
        'packaging-id': 'f7101ce3-0384-39ce-9fde-fbbd0044d35f',
        'text-representation': {
          'language': 'eng',
          'script': 'Latn'
        },
        'status': 'Official',
        'id': '053b7800-979f-4e7d-a15f-9f9fe2156687',
        'country': 'US',
        'barcode': null,
        'asin': 'B00SVNM7FO'
      },
      {
        'disambiguation': '',
        'cover-art-archive': {
          'back': false,
          'artwork': true,
          'front': true,
          'count': 1,
          'darkened': false
        },
        'packaging-id': 'f7101ce3-0384-39ce-9fde-fbbd0044d35f',
        'text-representation': {
          'script': 'Latn',
          'language': 'eng'
        },
        'status': 'Official',
        'title': 'Queen',
        'quality': 'normal',
        'release-events': [
          {
            'date': '1973-09-04',
            'area': {
              'id': '489ce91b-6658-3307-9877-795b68554c98',
              'iso-3166-1-codes': [
                'US'
              ],
              'name': 'United States',
              'disambiguation': '',
              'sort-name': 'United States'
            }
          }
        ],
        'status-id': '4e304316-386d-3409-af2e-78857eec5cfe',
        'packaging': 'Cardboard/Paper Sleeve',
        'date': '1973-09-04',
        'barcode': '',
        'asin': null,
        'id': '1783da6a-9315-3602-a488-1738eb733a0f',
        'country': 'US'
      }
    ]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
