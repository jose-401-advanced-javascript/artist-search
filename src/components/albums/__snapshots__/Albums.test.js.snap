// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Albums component renders Albums 1`] = `
<Fragment>
  <button
    type="button"
  >
    Back
  </button>
  <button
    type="button"
  >
    Next
  </button>
  <ul>
    <li
      key="053b7800-979f-4e7d-a15f-9f9fe2156687"
    >
      <AlbumCard
        asin="B00SVNM7FO"
        barcode={null}
        country="US"
        cover-art-archive={
          Object {
            "artwork": true,
            "back": false,
            "count": 2,
            "darkened": false,
            "front": false,
          }
        }
        date="1974-02-14"
        disambiguation=""
        id="053b7800-979f-4e7d-a15f-9f9fe2156687"
        packaging="Cardboard/Paper Sleeve"
        packaging-id="f7101ce3-0384-39ce-9fde-fbbd0044d35f"
        quality="normal"
        release-events={
          Array [
            Object {
              "area": Object {
                "disambiguation": "",
                "id": "489ce91b-6658-3307-9877-795b68554c98",
                "iso-3166-1-codes": Array [
                  "US",
                ],
                "name": "United States",
                "sort-name": "United States",
              },
              "date": "1974-02-14",
            },
          ]
        }
        status="Official"
        status-id="4e304316-386d-3409-af2e-78857eec5cfe"
        text-representation={
          Object {
            "language": "eng",
            "script": "Latn",
          }
        }
        title="Liar"
      />
    </li>
    <li
      key="1783da6a-9315-3602-a488-1738eb733a0f"
    >
      <AlbumCard
        asin={null}
        barcode=""
        country="US"
        cover-art-archive={
          Object {
            "artwork": true,
            "back": false,
            "count": 1,
            "darkened": false,
            "front": true,
          }
        }
        date="1973-09-04"
        disambiguation=""
        id="1783da6a-9315-3602-a488-1738eb733a0f"
        packaging="Cardboard/Paper Sleeve"
        packaging-id="f7101ce3-0384-39ce-9fde-fbbd0044d35f"
        quality="normal"
        release-events={
          Array [
            Object {
              "area": Object {
                "disambiguation": "",
                "id": "489ce91b-6658-3307-9877-795b68554c98",
                "iso-3166-1-codes": Array [
                  "US",
                ],
                "name": "United States",
                "sort-name": "United States",
              },
              "date": "1973-09-04",
            },
          ]
        }
        status="Official"
        status-id="4e304316-386d-3409-af2e-78857eec5cfe"
        text-representation={
          Object {
            "language": "eng",
            "script": "Latn",
          }
        }
        title="Queen"
      />
    </li>
  </ul>
</Fragment>
`;
